### DISCUSSION TITLE
"Broken pipe or closed connection" on $this->connection->channel()

### DISCUSSION THREAD

**Discussion by harpreetsb** [php-amqplib/php-amqplib#1149] [Q&A] [answered]
Hi,
I have been working on a personal project using Codeigniter4 (PHP 8.2), "php-amqplib/php-amqplib": "^3.5" on fedora 39
This is not the first time I am using this library.
The setup is simple and rabbitMQ service is running.
![image](https://github.com/php-amqplib/php-amqplib/assets/6584304/1d13501c-bb06-4d1d-b873-cdf929204b21)
```
$this->qConf        = new \Queue\Config\QueueConf();
            $this->exchangeName = 'router';
            $this->queueName    = 'msgs';

            $this->connection = new AMQPStreamConnection(
                $this->qConf->host,
                $this->qConf->port,
                $this->qConf->user,
                $this->qConf->password,
                $this->qConf->vhost
            );
           //===== no exception this here ====

            $this->channel = $this->connection->channel(); // <-- THIS THROWS EXCEPTION
```
![image](https://github.com/php-amqplib/php-amqplib/assets/6584304/a4b8068e-2adc-4e82-b95a-90f5cfaa2c4b)

Please help

**Comment by lukebakken** [comment]
We need the following information:
* What version of RabbitMQ and Erlang?
* What does RabbitMQ log at the same time as "Broken pipe or closed connection"?

**Reply by harpreetsb** [reply]
Version:
- with command `sudo rabbitmqctl --version` gives me `3.11.10`.
- with command `erl -eval 'erlang:display(erlang:system_info(otp_release)), halt().'  -noshell` gives me `26`
The log is same as [here](https://github.com/rabbitmq/rabbitmq-server/issues/9371)

**Reply by harpreetsb** [reply]
I think this may be due to, upgrade from Fedora 38 to 39.
That might have upgraded Erlang to 26, which is not supported by rabbitMQ 3.11.x.
Now what should I do next?

**Reply by lukebakken** [reply]
> That might have upgraded Erlang to 26, which is not supported by rabbitMQ 3.11.x. Now what should I do next?

You either need to upgrade RabbitMQ, or downgrade Erlang.

**Reply by harpreetsb** [reply]
I downloaded RPM x86_64 from 
https://koji.fedoraproject.org/koji/buildinfo?buildID=2335621
now rabbitmq wont start

**Reply by michaelklishin** [reply]
See server logs for clues. If this was an upgrade to 3.12, most likely [you haven't had all feature flags enabled before upgrading](https://rabbitmq.com/upgrade.html).

**Reply by michaelklishin** [reply]
Our Erlang RPM (both the Fedora 38 and the CentOS Stream 9 versions) supports Fedora 39 just fine.

If this is a development environment, your can remove and purge your installation (that is, its [data directory](https://rabbitmq.com/relocate.html)) then install Erlang 26.2 and RabbitMQ 3.12.11.

If you cannot afford to do that, you must install Erlang 25 and RabbitMQ 3.11.28, then enable **all** feature flags, then upgrade to Erlang 26 and RabbitMQ 3.12.11.

**Reply by michaelklishin** [reply, marked as answer]
Our [RPM distribution guide](https://rabbitmq.com/install-rpm.html) explains what repositories you should use if you want the latest stuff. Fedora does not provide RabbitMQ 3.12 (or even the latest 3.11) in its repositories (other than Rawhide) but it [does ship with Erlang 26](https://packages.fedoraproject.org/pkgs/erlang/erlang/).

Just nuke the RabbitMQ and Erlang packages installed from Fedora repositories and follow Team RabbitMQ's RPM guide, please.

**Comment by michaelklishin** [comment]
Apparently the Fedora-packaged RabbitMQ 3.11.10 RPM [assumes that it supports Erlang 26](https://packages.fedoraproject.org/pkgs/rabbitmq-server/rabbitmq-server/fedora-39.html) but that's [not the case](https://rabbitmq.com/which-erlang.html).

RabbitMQ 3.12.x does support Erlang 26 but 3.12.10 is only available via Fedora 39 testing repositories or in Fedora Rawhide.

In any case, a better solution is the same as for other operating systems: [use the repositories maintained by Team RabbitMQ](https://rabbitmq.com/install-rpm.html).

**Reply by harpreetsb** [reply]
Thank you so much.
I did following(as suggested) to get this running.

1. Uninstalled Erlang and RabbitMQ
2. deleted all directories as mentioned
3. Re-installed with the guide provided for latest fedora
All is working fine.

**Reply by michaelklishin** [reply]
RabbitMQ [Homebrew installation guide](https://rabbitmq.com/install-homebrew.html) now mentions feature flags and how to reinstall a node from scratch.

**Reply by harpreetsb** [reply]
I have another problem now
after running `sudo dnf update`. I get the following response:
![image](https://github.com/php-amqplib/php-amqplib/assets/6584304/b5a02844-e952-4e66-9f66-923c297ebf5a)
I am guessing this is related. 
help required
